public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/DimensionFields.tsx

```diff
@@ -48,12 +107,44 @@ const DimensionFields: React.FC<DimensionFieldsProps> = ({ query, dimensionOptio
               menuShouldPortal
               placeholder="Field"
               value={filter.dimension}
-              options={dimensionOptions}
+              options={validDimensionOptions}
               onChange={(v) => onFieldChange(index, 'dimension', v.value ?? '')}
               width={38}
             />
-            <InlineLabel aria-label="equals">==</InlineLabel>
-            <Input placeholder="" value={filter.filter} onChange={(ev) => onFilterInputChange(index, ev)} />
+            <Select
+              menuShouldPortal
+              placeholder="Operation"
+              value={filter.operator}
+              options={dimensionOperators}
+              onChange={(v) => onFieldChange(index, 'operator', v.value ?? '')}
+              allowCustomValue
+            />
+            {filter.operator === 'eq' || filter.operator === 'ne' ? (
+              <MultiSelect
```

Comment:
Can you add unit tests for the conditional rendering?

Reply:
Will be moved to a separate PR